### ISSUE TITLE
Crash when archiving CZWeatherCurrentCondition, CZWeatherForecastCondition, CZWeatherHourlyCondition & CZWeatherLocation

### ISSUE THREAD

**Issue by awph** [comyarzaheri/CZWeatherKit#32]
When archiving a CZWeatherCurrentCondition, CZWeatherForecastCondition, CZWeatherHourlyCondition or CZWeatherLocation object the application crash. The crash is due to a bug when archiving structs.

Execution of:

```objective-c
[NSKeyedArchiver archiveRootObject:weatherData toFile:[NSTemporaryDirectory() stringByAppendingString:@"archive"]];
```

Result in the following exception:

```
Assertions: failed: caught "NSInvalidArgumentException", "*** -[NSKeyedArchiver encodeValueOfObjCType:at:]: this archiver cannot encode structs"
(
	0   CoreFoundation                      0x000000010cc1df45 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x000000010c697deb objc_exception_throw + 48
	2   CoreFoundation                      0x000000010cc1de7d +[NSException raise:format:] + 205
	3   Foundation                          0x000000010c302a32 -[NSKeyedArchiver encodeValueOfObjCType:at:] + 311
	4   Foundation                          0x000000010c329b4a -[NSValue encodeWithCoder:] + 317
	5   Foundation                          0x000000010c328f22 -[NSConcreteValue encodeWithCoder:] + 297
	6   Foundation                          0x000000010c2916c7 _encodeObject + 1259
	7   CZWeatherKit                        0x00000001167380e8 -[CZWeatherLocation encodeWithCoder:] + 284
	8   Foundation                          0x000000010c2916c7 _encodeObject + 1259
	9   Foundation                          0x000000010c2cc442 +[NSKeyedArchiver archiveRootObject:toFile:] + 241
	...
)
```

The part the cause this crash is:
```objective-c
NSValue *coordinateValue = [NSValue value:&_coordinate
                                 withObjCType:@encode(CLLocationCoordinate2D)];
[aCoder encodeObject:coordinateValue forKey:@"coordinate"];
```
It's correct to do this, but it's seams to be a knowing bug ([rdar://8631663](http://www.openradar.me/8631663)).

An existing workaround is to archive/unarchive the values of the struct manually.

Something like this: 
```objective-c
NSNumber *latitude = [NSNumber numberWithDouble:self.coordinate.latitude];
NSNumber *longitude = [NSNumber numberWithDouble:self.coordinate.longitude];
[aCoder encodeObject:latitude forKey:@"latitude"];
[aCoder encodeObject:longitude forKey:@"longitude"];
```

The problem is exactly the same with the structs CZHumidity, CZTemperature and CZWindDirection.

I'm going to fix it on my side, and create a PR.

Also if you know any better solution, said it!

**Comment by awph** [comment]
This PR also fix a bug where windDirection was set instead of windSpeed [here](https://github.com/comyarzaheri/CZWeatherKit/blob/b0cfa72165a22722119927b7af92df9c1d26ff42/CZWeatherKit/CZWeatherCurrentCondition.m#L76) and [here](https://github.com/comyarzaheri/CZWeatherKit/blob/b0cfa72165a22722119927b7af92df9c1d26ff42/CZWeatherKit/CZWeatherHourlyCondition.m#L72).

Adding tests for these classes would be :+1:

**Comment by comyarzaheri** [comment]
Resolved by #33, thanks!
